### ISSUE TITLE
Circular refs hang on Windows

### ISSUE THREAD

**Issue by fredbi** [go-openapi/spec#64]
Unit test CircularRefsExpansion fails to detect circular ref and hang on Windows,
possibly because of #63.

```
C:\Users\root\go\src\github.com\go-openapi\spec>go test -v -run CircularRefsExpansion
=== RUN TestCircularRefsExpansion
```

When looking for circular reference given by non-normalized Ref at https://github.com/go-openapi/spec/blob/1de3e0542de65ad8d75452a595886fdd0befb363/expander.go#L621

the circular ref check fails to identify that the ref has already been encountered

**Comment by mtfelian** [comment]
Fixed this in PR #67

**Comment by fredbi** [comment]
Verified. Works. Thanks for the great job
